### ISSUE TITLE
Flow 2-Job 2: `1b-upsertCases.js` is updating the wrong record; `record_id` mapping not outputting expected id

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/unicef-thailand#79]
## Describe the bug
A primero user clicked the `Sync` button in Primero that triggered the webhook to send [this Message](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/messages/0635f2b0-18c6-7810-b942-c2ba5c2a4f23) to OpenFn with these case details: 
`"national_id_no": "1209501299178"`
`"record_id": "12da71a1-dc70-42b3-9496-ce8f4de05a1f"`

However, when [this job 2 runs](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0636249f-950b-717f-a4f7-1668199a94f8), it is NOT using the `record_id` received in the triggering Message:
```
Case found. Performing update.
PATCH succeeded: 200 OK
{
  id: 'db77e14e-966e-44bc-b162-3f319f21959a' 
```
^this `id` SHOULD = `'12da71a1-dc70-42b3-9496-ce8f4de05a1f'` which is the `record_id` identified in [L25 in the message](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/messages/0635f2b0-18c6-7810-b942-c2ba5c2a4f23) that triggered this whole flow. 

It _looks_ like the job is correctly upserting the case using the `record_id` as the lookup key, so why are we mapping a different `record_id` than received in the triggering Message? 
https://github.com/OpenFn/primero-thailand/blob/8b50af87323e7cb3a56bc31c3d824a833e43ec9d/jobs/1b-upsertCases.js#L3480-L3488

**NOTE that we identified that there are 2 cases in Primero with duplication `national_id_no: 1209501299178` .. this _should_ not happen in real life (and we have flagged to the client), but given that we use `record_id` as the externalId upsert key on [L3482](https://github.com/OpenFn/primero-thailand/blob/staging/jobs/1b-upsertCases.js#L3482) of the job, I don't understand why we're encountering issues here... because we have the right `record_id` to upsert on, but don't seem to be using it.  Therefore I wonder if there is an issue with the adaptor's helper function. 

To recap relevant unique identifiers: 
1. `record_id` or `id` is the unique system-generated ID for the case record in Primero. It appears in the URL when accessing a record via the web UI. 
2. `case_id` is a second unique system-generated ID for the case record in Primero. This is a more business-friendly ID that appears on the record page of a case record when viewing in the web UI. 
3. `national_id_no` _should_ be unique for each case, but in this test demo environment, we are finding some scenarios where multiple cases have the same national id. 

## To Reproduce
1. Run job 1 `/staging/jobs/1-getPatientData.js` with this Message as state.data: https://www.openfn.org/projects/primero-thailand-io-unicef---staging/messages/0635f2b0-18c6-7810-b942-c2ba5c2a4f23
4. Then run job 2 `staging/jobs/1b-upsertCases.js` with the final state from this Job 1 run. The run will look like this: https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0636249f-950b-717f-a4f7-1668199a94f8

We want to troubleshoot the `record_id` mapping because L8 of this^ run log should show ` id: '12da71a1-dc70-42b3-9496-ce8f4de05a1f'`

### expression.js
*Please make changes in the `staging` branch!* ⚠️
Link to the job itself in Github: https://github.com/OpenFn/primero-thailand/blob/staging/jobs/1b-upsertCases.js

#### adaptor
`language-primero@2.9.1`

### state.json
This is step 2 job in a flow. 
1. Run job1 `/staging/jobs/1-getPatientData.js` to generate state for job 2. 
- state.configuration: `MOPH Thailand HIS API - DEMO Alpha (Unicef)`
- state.data: Use this Message: https://www.openfn.org/projects/primero-thailand-io-unicef---staging/messages/0635f2b0-18c6-7810-b942-c2ba5c2a4f23
5.  Use the final state of the first job to run the 2nd job
- `state.configuration`: see LP `Primero Thailand Demo API User (Open Function)` (Note... this user is for **basic auth**.)

## Expected behavior
We should be updating the case with record `id: '12da71a1-dc70-42b3-9496-ce8f4de05a1f'` that triggered the flow. 

## To test/resolve
1. After the desired output is working locally, please push commits to `staging`.
2. I will then re-run [this job run](https://www.openfn.org/projects/primero-thailand-io-unicef---staging/runs/0636249e-ef0a-7377-a3f2-cc3d63d1d697) to re-trigger the flow on platform. 

## Toggl 
UNICEF Thailand

**Comment by aleksa-krolls** [comment]
@mtuchi can you do an initial triage here before EOD? I'm worried there's a primero adaptor issue with the `upsertCase(...)` function

**Comment by mtuchi** [comment]
> @mtuchi can you do an initial triage here before EOD? I'm worried there's a primero adaptor issue with the `upsertCase(...)` function

This has been fixed on #80

**Comment by jayesse** [comment]
Thanks @mtuchi. Closing this issue now.
